ibis/udf/vectorized.py

```diff
@@ -38,7 +42,44 @@ def __call__(self, *args, **kwargs):
         # kwargs.
         @functools.wraps(self.func)
         def func(*args):
-            return self.func(*args, **kwargs)
+            # If cols are pd.Series, then we save and restore the index.
+            if hasattr(args[0], 'index'):
+                saved_index = args[0].index
+            else:
+                saved_index = None
+
+            result = self.func(*args, **kwargs)
+
+            if isinstance(self.output_type, dt.Struct):
+                if isinstance(result, pd.DataFrame) or (
+                    isinstance(result[0], (pd.Series, list, np.ndarray))
```

Comment:
does a 0-len return type hit here? (it would error), tests?

Reply:
Functionally this should never happen because it's guarded by 
```
if isinstance(self.output_type, dt.Struct)
```
which by definition means the output should not be 0-len.